gdal/port/cpl_vsil_curl.cpp

```diff
@@ -4216,6 +4216,22 @@ char** VSICurlFilesystemHandler::ReadDirEx( const char *pszDirname,
     return ReadDirInternal(pszDirname, nMaxFiles, nullptr);
 }
 
+/************************************************************************/
+/*                             SiblingFiles()                           */
+/************************************************************************/
+
+char** VSICurlFilesystemHandler::SiblingFiles( const char *pszFilename )
+{
+    /* Small optimization to avoid unnecessary stat'ing from PAux or ENVI */
+    /* drivers. The MBTiles driver needs no companion file. */
+    if( EQUAL(CPLGetExtension( pszFilename ),"mbtiles") )
+    {
+        return static_cast<char**> (calloc(1,sizeof(char*)));
```

Comment:
the previous hack returned `CSLAddString( nullptr, CPLGetFilename(pszFilename) )` instead of an empty list. Does the list of sibling files also need to include the main file?

Reply:
I *guess* the empty list shall be enough. Let's see if test are happy

Reply:
Local testing here showed no issues :crossed_fingers: